.circleci/config.yml

```diff
@@ -0,0 +1,81 @@
+version: 2
+
+job_default: &job_defaults
+  working_directory: ~/vscode-wsdl2rest
+  docker:
+    - image: circleci/openjdk:8-jdk-node-browsers
+
+common_env: &common_env
+  MAVEN_OPTS: -Xmx512m
+
+jobs:
+  build:
+    <<: *job_defaults
+    environment:
+      <<: *common_env
+    steps:
+      - checkout
+      - restore_cache:
+          key: dependency-cache-{{ checksum "package-lock.json" }}
+      - run:
+          name: install-typescript
+          command: sudo npm install -g typescript
+      - run:
+          name: install-mocha
+          command: sudo npm install -g mocha
+      - run:
+          name: install-vsce
+          command: sudo npm install -g vsce
+      - run:
+          name: npm-ci
+          command: npm ci
+      - run:
+          name: vsce-package
+          command: vsce package
+      - run:
+          name: test
+          command: npm test
+      - run:
+          name: test UI
+          command: npm run ui-test
+      - store_artifacts:
+          path: ./wsdl2rest.log
+      - save_cache:
+          key: dependency-cache-{{ checksum "package-lock.json" }}
+          paths:
+            - ~/.m2
+            - ~/.npm
```

Comment:
given that the cache is also for .m2,, i think we need to use a key using a checksum of both package.json and pom.xml

Reply:
good point. I will adapt that logic a bit.